### PR TITLE
API clients may use the HTTP Authorization header for auth. 

### LINKED ISSUE #1642: HTTP API should allow JWT to be passed via Authorization header
It is currently necessarily to pass the auth JWT via a session cookie in order to interact with the Argo CD HTTP API. IMO this is not typical. Instead (or in addition) the HTTP API should accept the auth JWT via an `Authorization` header. E.g. `Authorization: Bearer <token>`. This will, at least partially, mitigate #1103.

Switching to header-only will introduce problems with API calls sent from HTML `link` hrefs, because the browser will send the session cookie in those instances, but not set the `Authorization` header. Again in my opinion, that is a fairly non-standard way to interact with an HTTP API. For that use case, I would suggest accepting the JWT via a query param instead.

### PR DESCRIPTION
**Pull request by alexec**
Closes #1642

### CODE REVIEW

**Comment by alexec** [comment]
`TestAddProjectSource` failed, re-build.

File: server/server.go [hunk 1]
```diff
@@ -753,22 +753,39 @@ func (a *ArgoCDServer) getClaims(ctx context.Context) (jwt.Claims, error) {
 
 // getToken extracts the token from gRPC metadata or cookie headers
 func getToken(md metadata.MD) string {
-	// check the "token" metadata
-	tokens, ok := md[apiclient.MetaDataTokenKey]
-	if ok && len(tokens) > 0 {
-		return tokens[0]
+	// check the "t" metadata
+	log.Debugf("ALEX md=%v", md)
```
Comment (jessesuen) [inline server/server.go]: Please remove

**Review by jessesuen — APPROVED** [review]
Conditional approval after removing `ALEX` debug code.

File: server/server.go (hunk 1 above)
Comment (alexec) [reply server/server.go]: ops!

**Pull request merged**